tests/ui/map_unwrap_or_else_fixable.fixed

```diff
@@ -0,0 +1,59 @@
+// run-rustfix
+// aux-build:option_helpers.rs
+
+#![warn(clippy::map_unwrap_or)]
+
+#[macro_use]
+extern crate option_helpers;
+
+use std::collections::HashMap;
+
+#[rustfmt::skip]
+fn option_methods() {
+    let opt = Some(1);
+
+    // Check for `option.map(_).unwrap_or_else(_)` use.
+    // single line case
+    let _ = opt.map_or_else(|| 0, |x| x + 1);
+
+    // Macro case.
+    // Should not lint.
+    let _ = opt_map!(opt, |x| x + 1).unwrap_or_else(|| 0);
+
+    // Check for `option.map(_).unwrap_or_else(_)` use.
+    // single line case
+    let _ = opt.map_or_else(|| 0, |x| x + 1);
+
+    // Issue #4144
+    {
+        let mut frequencies = HashMap::new();
+        let word = "foo";
+
+        frequencies
+            .get_mut(word)
+            .map(|count| {
+                *count += 1;
+            })
+            .unwrap_or_else(|| {
+                frequencies.insert(word.to_owned(), 1);
+            });
```

Comment:
>I wonder why multi-line cases are not giving suggestions.

Are you refering to this test? I think this is expected since #4164 avoids suggesting something that would mutably borrow twice.

Reply:
I was talking about this kind of multi line case: https://github.com/rust-lang/rust-clippy/pull/6197/files#diff-3006faeb70e68a305c034790612632b755f8dc92655969a36714aabe70c77648R66

Reply:
So I traced this back to [this comment](https://github.com/rust-lang/rust-clippy/pull/468#discussion_r45575391). It seems to be to avoid giving huge suggestions.

I would say something improved since then and the reporting infrastructure shows it someway shortened, but the automatic fix is correctly applied.

Now that we're at it, would you mind checking if that's the case? Otherwise we can leave it like that and merge the PR :)

Reply:
I tried something meaningless:
```
let _ = res.map(|x| x + 1)
        .unwrap_or_else(|_e| {
            let a = 2;
            let b = 1;
            let mut total = 0;
            total += 1;
            total *= b;
            total = total + a + b;
            total = if a == 2 { 
                a
            } else {
                b
            };
            total
        });
```
stderr:
```  
   |
LL |       let _ = res.map(|x| x + 1)
   |  _____________^
LL | |         .unwrap_or_else(|_e| {
LL | |             let a = 2;
LL | |             let b = 1;
...  |
LL | |             total
LL | |         });
   | |__________^
   |
help: try this
   |
LL |     let _ = res.map_or_else(|_e| {
LL |             let a = 2;
LL |             let b = 1;
LL |             let mut total = 0;
LL |             total += 1;
LL |             total *= b;
 ...
```

The result is not that good, as the user doesn't have the full code in the `try this` part, so you can't fix it by hand (though auto-fix will work anyway), but it's also not that bad, because in this lint the important part is at the beginning.

Reply:
Thanks for trying it!

So in general I think we should provide auto-fixable suggestions if it's feasible, but as this was already special cased and the presentation is still far from perfect, let's leave it like this for now.